### ISSUE TITLE
convenience playbooks fail to missing facts

### ISSUE THREAD

**Issue by Underknowledge** [ansible-collections/community.sops#145]
##### SUMMARY

Coming from https://docs.ansible.com/ansible/latest/collections/community/sops/docsite/guide.html

the Task `Install system packages` 
https://github.com/ansible-collections/community.sops/blob/b7bfe2a78660de3ceb37f3ccc84c3255ef007fa1/roles/install/tasks/main.yml#LL68C5-L68C5
Fails to the missing facts gathering here
https://github.com/ansible-collections/community.sops/blob/b7bfe2a78660de3ceb37f3ccc84c3255ef007fa1/playbooks/install_localhost.yml#LL7C22-L7C22
Here I'm unsure if the Documentation or the playbook should be updated. 
`setting the \"use\" option` option could maybe be done, not not yet seen this option. 
  
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

task

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.1]
  config file = None
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/user/.local/bin/ansible
  python version = 3.10.11 (main, Apr  5 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)]
  jinja version = 3.0.3
  libyaml = False
```

##### COLLECTION VERSION

```paste below
# /home/user/.local/lib/python3.10/site-packages/ansible_collections
Collection     Version
-------------- -------
community.sops 1.2.2  

# /home/user/.ansible/collections/ansible_collections
Collection     Version
-------------- -------

```

##### CONFIGURATION

```paste below
---
```

##### OS / ENVIRONMENT

Fedora 38 

##### STEPS TO REPRODUCE

```yaml
ansible-playbook community.sops.install_localhost
```

##### EXPECTED RESULTS

No failure

##### ACTUAL RESULTS

```paste below
TASK [community.sops.install : Install system packages] **************************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NoneType: None
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Could not detect which package manager to use. Try gathering facts or setting the \"use\" option."}
```

**Comment by felixfontein** [comment]
Thanks for your report! I created a PR (#146) to fix this. Could you please test it?

**Comment by felixfontein** [comment]
Note that since Ansible 2.0.0, `package` calls the `setup` module itself to gather the required package manager fact (https://github.com/ansible/ansible/commit/172e0c328c0d9d1e53fe1557f8504b908598d3b9). If that fails for some reason, the error message you see is printed (`Could not detect which package manager to use. Try gathering facts or setting the "use" option.`). The code is still there in only a slightly modified version in the devel version of ansible-core (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/package.py#L56).

So basically if you get this message, #146 won't likely help you either. It seems that facts gathering doesn't work (at least for `pkg_mgr`) on your machine. You need to figure out what is preventing facts gathering to work.

**Comment by Underknowledge** [comment]
Nah, would state that it works with your proposed change. 

```log
ansible-playbook community.sops.install_localhost -K
BECOME password: 
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: running playbook inside collection community.sops

PLAY [Install sops on localhost] *****************************************************************************************************************************************************************************

TASK [Install sops on localhost] *****************************************************************************************************************************************************************************

TASK [community.sops.install : Validating arguments against arg spec 'main' - Install Mozilla sops] **********************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Gather required information on localhost] *************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Show system information] ******************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Architecture: x86_64\nDistribution: Fedora 38\nDistribution version: 38\nOS family: RedHat\npkg mgr: dnf"
}

TASK [community.sops.install : Include distribution specific variables] **************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Start determining source] *****************************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Auto-detect source to install sops from] **************************************************************************************************************************************
included: /home/user/.ansible/collections/ansible_collections/community/sops/roles/install/tasks/detect_source.yml for localhost

TASK [community.sops.install : Check whether system packages are a valid source of sops latest] **************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Check whether GitHub is a valid source of sops] *******************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Ensure that something was detected] *******************************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Install Mozilla sops from GitHub] *********************************************************************************************************************************************
included: /home/user/.ansible/collections/ansible_collections/community/sops/roles/install/tasks/github.yml for localhost

TASK [community.sops.install : Make sure that sops can be installed from GitHub] *****************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Start determining sops version] ***********************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Determine latest version (fallback)] ******************************************************************************************************************************************
included: /home/user/.ansible/collections/ansible_collections/community/sops/roles/install/tasks/github_api.yml for localhost

TASK [community.sops.install : Fetch list of releases from GitHub] *******************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : In case rate limit was exceeded, inform user] *********************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Determine the latest release] *************************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Determine latest version] *****************************************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Fail when latest version could not be selected] *******************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Show selected version] ********************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "The latest sops version is sops 3.7.3."
}

TASK [community.sops.install : Set variables] ****************************************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Install Mozilla sops from system package repositories] ************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Install system packages] ******************************************************************************************************************************************************
ok: [localhost]

TASK [community.sops.install : Install unsigned system packages] *********************************************************************************************************************************************
changed: [localhost]

TASK [community.sops.install : Install packages from URL/path (Debian)] **************************************************************************************************************************************
skipping: [localhost]

TASK [community.sops.install : Set results] ******************************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP ***************************************************************************************************************************************************************************************************
localhost                  : ok=17   changed=1    unreachable=0    failed=0    skipped=8    rescued=0    ignored=0   

~ took 23s ❯ 
```

For completness I would add the Package manager to the debug output

```yaml
    - name: Show system information
      ansible.builtin.debug:
        msg: |-
          Architecture: {{ _community_sops_install_facts.architecture }}
          Distribution: {{ _community_sops_install_facts.distribution }} {{ _community_sops_install_facts.distribution_major_version }}
          Distribution version: {{ _community_sops_install_facts.distribution_version }}
          OS family: {{ _community_sops_install_facts.os_family }}
          Package manager: {{ _community_sops_install_facts.pkg_mgr }}
 ```


maybe `gather_facts: false` on the playbook level blocks the detection from package? cant say. 
Not closing yet as you might still want to merge https://github.com/ansible-collections/community.sops/pull/146

**Comment by felixfontein** [comment]
Now I'm really curious why calling setup itself from the package action doesn't work.

**Comment by Underknowledge** [comment]
Could maaaaybe be related, when getting the error `Try gathering facts or setting the "use" option.` I had not provided a become password.

**Comment by felixfontein** [comment]
Can you try again without providing a `become` password but with #146? I'm wondering whether #146 makes a difference, or whether it will fail with and without #146.

**Comment by Underknowledge** [comment]
So sorry for the late reply, written something and then never pressed on send. 
With the change, the task just gracefully fails, telling you that you don't have the needed privilege or no password is provided.
